tools/autopep8_extended.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import ast
+import re
+import sys
+
+import autopep8
+import inflection
+
+
+class Pep8Extended(object):
+    def __init__(self, pep8_options, source):
+        self.pep8_options = pep8_options
+        self.source = source
+
+    def get_checks(self):
+        msgs = {
+            'CW0001': 'Class name with snake_case style found, '
+                      'should use CamelCase. Change of "{old_class_name}" '
+                      'to "{new_class_name}" next (lines,columns): '
+                      '{lines_columns}',
+        }
+        return msgs
+
+    def check_cw0001(self):
+        code = 'CW0001'
+        msg = self.get_checks()[code]
+        class_renamed = {}
+        check_result = []
+
+        # Fix 'SyntaxError: encoding declaration in Unicode string'
+        parsed_source = self.source
+        line_deleted = False
+        if len(parsed_source) >= 1 and \
+           ('# -*- coding:' in parsed_source[0] or
```

Comment:
Not sure if you cover all the ways you can declare the encoding.. Moreover the detection logic is duplicated...
What do you think of
```python
lines = parsed_source[0:2]
for index, line in enumerate(lines):
    if line.startswith('#') and 'conding:' in line:
        line_deleted = 1
        parsed_source = parsed_source[0:index] + parsed_source[index+1:]
```